### PR TITLE
ExtensionApp: change extension_name to name.

### PR DESCRIPTION
**Pull request by Zsailer**
Changes the `extension_name` trait to `name` in ExtensionApp. 

This addresses issues in #222 and #224. @echarles

### CODE REVIEW

**Comment by echarles** [comment]
Thx @Zsailer. So for now this PR doesn't pick the python extension standard mecanism but we stick with the `_jupyter_server_extension_paths`?

I have pulled the changes and updated the example. I fail with 

```
  File "/Users/echar4/datalayer/repos/jupyter-server/examples/simple/simple_ext1/application.py", line 48, in initialize_handlers
    (r'/{}/default'.format(self.extension_name), DefaultHandler),
AttributeError: 'SimpleApp1' object has no attribute 'extension_name'
```

**Comment by Zsailer** [comment]
That’s right, @echarles. I think the entry point discussion can be separated from this for now. We’ll discuss that at our next server meeting. 

I’ll update the examples directory next

**Comment by echarles** [comment]
Cool! Having a way to create e.g. jupyter_lab_config,.py will be an excellent first step (PS: I won't be able to join today meeting)

**Comment by Zsailer** [comment]
I've updated the examples (sorry for the delay, here).

This should be ready for merge.

**Comment by echarles** [comment]
Thx Zach. There are still some `extension_name` used, at least here.

https://github.com/Zsailer/jupyter_server/blob/ff7826636e9a47665eb6a99ae1d28982441941e6/examples/simple/simple_ext1/application.py#L48-L53

BTW As discussed, I will open a PR to run the examples by GitHub Actions.

File: examples/simple/simple_ext1/handlers.py [hunk 1]
```diff
@@ -4,17 +4,17 @@
 class DefaultHandler(ExtensionHandlerMixin, JupyterHandler):
     def get(self):
         # The name of the extension to which this handler is linked.
-        self.log.info("Extension Name in {} Default Handler: {}".format(self.extension_name, self.extension_name))
+        self.log.info("Extension Name in {} Default Handler: {}".format(self.name, self.name))
         # A method for getting the url to static files (prefixed with /static/<extension_name>).
```
Comment (Zsailer) [inline examples/simple/simple_ext1/handlers.py:8]:
```suggestion
        # A method for getting the url to static files (prefixed with /static/<name>).
```

Comment (Zsailer) [reply examples/simple/simple_ext1/handlers.py]: Fixed

**Comment by Zsailer** [comment]
Thanks, @echarles. Merging!

**Comment by echarles** [comment]
@Zsailer I have updated the jlab branches to latest master including this PR.

It works, but when I change `name=jupyterlab` to `name=lab`, I receive exeception

```
  File "/Users/echar4/datalayer/repos/jupyter-server/jupyter_server/serverapp.py", line 1534, in init_server_extensions
    mod, metadata_list = _get_server_extension_metadata(module_name)
  File "/Users/echar4/datalayer/repos/jupyter-server/jupyter_server/extension/serverextension.py", line 43, in _get_server_extension_metadata
    m = import_item(module)
  File "/opt/datalayer/opt/miniconda3/envs/datalayer/lib/python3.7/site-packages/traitlets/utils/importstring.py", line 43, in import_item
    return __import__(parts[0])
ModuleNotFoundError: No module named 'lab'
```

I have nailed down the issue but we need to discuss about that.

**Comment by echarles** [comment]
@Zsailer for example, if you change in the simple_ext1 the name and you say `name = 'ext1'`, the server launches fine with `jupyter server --ServerApp.jpserver_extensions="{'ext1': True}"` but fails with `python -m simple_ext1` or launched via the entrypoints.

**Pull request merged**